llvm/lib/TargetParser/Triple.cpp

```diff
@@ -833,8 +831,10 @@ static Triple::ObjectFormatType getDefaultFormat(const Triple &T) {
       return Triple::COFF;
     else if (T.isUEFI())
       return Triple::COFF;
-    return Triple::ELF;
 
+    if (T.isOSDarwin())
```

Comment:
nit: should we combine this with the two if's above and make it a switch on the OS (Windows, UEFI and Darwin are all OS bits)?

Reply:
It kind of works but isOSDarwin covers a few too many things:

`    return isMacOSX() || isiOS() || isWatchOS() || isDriverKit();`
```
  bool isiOS() const {
    return getOS() == Triple::IOS || isTvOS();
  }
```